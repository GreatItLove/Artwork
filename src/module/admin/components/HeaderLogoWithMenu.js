import React from 'react';
import { Link } from 'react-router-dom';
import FloatingActionButton from 'material-ui/FloatingActionButton';
import PropTypes from 'prop-types';

import logo from '../../../img/logo-artwork-manager-white.png';
import { ADMIN_PANEL_BASE_PATH } from '../../../config/routesPatch';

const propTypes = {
  toggleMenu: PropTypes.func.isRequired
};

/**
 * @function HeaderLogoWithMenu
 * @param {Function} toggleMenu
 * @returns {XML}
 */
function HeaderLogoWithMenu({ toggleMenu }) {
  return (
    <div className="an-header">
      <div className="header-left">
        <Link to={ADMIN_PANEL_BASE_PATH} className="brand">
          <img src={logo} alt="Managed ArtWork logo" />
        </Link>
        <FloatingActionButton mini className="icon-buttons" onClick={toggleMenu}>
          <i className="material-icons icon-brand-black">format_indent_decrease</i>
        </FloatingActionButton>
      </div>
    </div>
  );
}
HeaderLogoWithMenu.propTypes = propTypes;

export { HeaderLogoWithMenu as HeaderLogoWithMenuTest };

export default HeaderLogoWithMenu;
