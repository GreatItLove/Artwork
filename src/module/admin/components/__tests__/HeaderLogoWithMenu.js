/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { HeaderLogoWithMenuTest as HeaderLogoWithMenu } from '../HeaderLogoWithMenu';

describe('HeaderLogoWithMenu component', () => {
  it('Shallow snapshot', () => {
    const organization = 'demo galery';
    const toggleMenu = jest.fn();
    const wrapper = shallow(<HeaderLogoWithMenu {...{ organization, toggleMenu }} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
