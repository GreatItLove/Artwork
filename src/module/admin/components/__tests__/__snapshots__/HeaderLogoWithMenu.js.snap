// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLogoWithMenu component Shallow snapshot 1`] = `
<div
  className="an-header"
>
  <div
    className="header-left"
  >
    <Link
      className="brand"
      replace={false}
      to="/"
    >
      <img
        alt="Managed ArtWork logo"
        src="logo-artwork-manager-white.png"
      />
    </Link>
    <FloatingActionButton
      className="icon-buttons"
      disabled={false}
      mini={true}
      onClick={[Function]}
      secondary={false}
      zDepth={2}
    >
      <i
        className="material-icons icon-brand-black"
      >
        format_indent_decrease
      </i>
    </FloatingActionButton>
  </div>
</div>
`;
